rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateGradleWrapper.java

```diff
@@ -222,16 +219,26 @@ public Properties visitEntry(Properties.Entry entry, ExecutionContext ctx) {
                             return entry;
                         }
 
-                        // Typical example: https://services.gradle.org/distributions/gradle-7.4-all.zip
+                        // Typical example: https://services.gradle.org/distributions/gradle-7.4-all.zip or https://company.com/repo/gradle-8.2-bin.zip
                         String currentDistributionUrl = entry.getValue().getText();
 
                         GradleWrapper gradleWrpr = getGradleWrapper(ctx);
-                        if (StringUtils.isBlank(gradleWrpr.getDistributionUrl()) &&
-                            !StringUtils.isBlank(version)  && Semver.validate(version, null).getValue() instanceof ExactVersion) {
+                        if (StringUtils.isBlank(gradleWrpr.getDistributionUrl()) && !StringUtils.isBlank(version) &&
+                            Semver.validate(version, null).getValue() instanceof ExactVersion) {
                             String newDownloadUrl = currentDistributionUrl.replace("\\", "")
-                                    .replaceAll("(.*gradle-)(\\d+\\.\\d+(?:\\.\\d+)?)(.*-(?:bin|all).zip)", "$1" + gradleWrapper.getVersion() + "$3");
+                                    .replaceAll("(.*gradle-)(\\d+\\.\\d+(?:\\.\\d+)?)(.*-(?:bin|all).zip)",
+                                            "$1" + gradleWrapper.getVersion() + "$3");
                             gradleWrapper = new GradleWrapper(version, new DistributionInfos(newDownloadUrl, null, null));
                         }
+                        String wrapperHost = currentDistributionUrl.substring(0, currentDistributionUrl.lastIndexOf("/")) + "/gradle-";
+                        if (StringUtils.isBlank(wrapperUri) && !StringUtils.isBlank(gradleWrpr.getDistributionUrl()) &&
+                            !gradleWrpr.getPropertiesFormattedUrl().startsWith(wrapperHost)) {
+                            String newDownloadUrl = gradleWrpr.getDistributionUrl()
+                                    .replace("\\", "")
+                                    .replaceAll("(.*gradle-)(\\d+\\.\\d+(?:\\.\\d+)?)(.*-(?:bin|all).zip)",
+                                            wrapperHost + gradleWrapper.getVersion() + "$3");
+                            gradleWrapper = new GradleWrapper(gradleWrpr.getVersion(), new DistributionInfos(newDownloadUrl, null, null));
+                        }
```

Comment:
I don't think this section is necessary as the wrapperUri would already be what's used in the `GradleWrapper` object from the `getGradleWrapper()` invocation.

Reply:
That's not necessarily true i think. 

Imagine calling the updatewrapper("8.2", null...) with a custom url already present in the properties file. Getwrapper would return the gradle services hosted wrapper (as no custom url is given) with sha and jarsha. We want to early on replace that with the custom distribution url wrapper without shas so that later replacement could be removed in the properties entry visitor (as that was where the patch of url happened before) + would add a sha in the properties visitor (previous bug I discovered during adding tests)

Reply:
Ok, I see. The previous case is we weren't able to resolve while this case is we were. 

Any reason not to just merge these two of blocks together then? So that if the distribution host is null or doesn't match?

There's an additional edge case of going from internal to external hosting that could be another possibility.

Reply:
Well... I thought about it, but there's a small difference in the `replaceAll`. Merging would require to extract the `$1` from first if-block through custom code in order to end up with a single `newDownloadUrl` assignment. 
I thought that's not worth it.

Reply:
@shanman190 do you want me to try to merge the 2 blocks? Or can we close this thread and merge?

Reply:
I've thought about this as well and it's probably fine to leave it like you've got it presently.